tests/test_utils.py

```diff
@@ -27,3 +30,49 @@ def test_get_regular_chunks_max_inclusive():
     assert list(target[0]) == list(range(4, 5))
     # The last chunk should include 44
     assert list(target[-1]) == list(range(44, 45))
+
+
+def test_buffered_slice_writer():
+    test_img = np.arange(24 * 24).reshape(24, 24).astype(np.uint16) + 1
+    dtype = test_img.dtype
+    bbox = {'topleft': (0, 0, 0), 'size': (24, 24, 35)}
+    origin = [0, 0, 0]
+    dataset_dir = 'testoutput/buffered_slice_writer'
+    layer_name = 'color'
+    mag = Mag(1)
+    dataset_path = os.path.join(dataset_dir, layer_name, mag.to_layer_name())
+
+    with BufferedSliceWriter(dataset_dir, layer_name, dtype, bbox, origin, mag=mag) as writer:
+        for i in range(13):
+            writer.write_slice(i, test_img)
+        with wkw.Dataset.open(dataset_path, wkw.Header(dtype)) as data:
+            try:
+                read_data = data.read(origin, (24, 24, 13))
+                if read_data[read_data.nonzero()].size != 0:
+                    raise AssertionError('Nothing should be written on the disk. But found data with shape: {}'
+                                        .format(read_data.shape))
+            except wkw.wkw.WKWException:
+                pass
+
+        for i in range(13, 32):
+            writer.write_slice(i, test_img)
+        with wkw.Dataset.open(dataset_path, wkw.Header(dtype)) as data:
+            read_data = data.read(origin, (24, 24, 32))
+            assert np.squeeze(read_data).shape == (24, 24, 32), "The read data should have the shape: (24, 24, 32) " \
+                                                                "but has a shape of: {}"\
+                                                                .format(np.squeeze(read_data).shape)
+            assert read_data.size == read_data[read_data.nonzero()].size, "The read data contains zeros while the " \
+                                                                          "written image has no zeros"
+
+        for i in range(32, 35):
+            writer.write_slice(i, test_img)
+
+    with wkw.Dataset.open(dataset_path, wkw.Header(dtype)) as data:
+        read_data = data.read(origin, (24, 24, 35))
+        assert np.squeeze(read_data).shape == (24, 24, 35), "The read data should have the shape: (24, 24, 35) " \
+                                                                "but has a shape of: {}"\
+                                                                .format(np.squeeze(read_data).shape)
+        assert read_data.size == read_data[read_data.nonzero()].size, "The read data contains zeros while the " \
+                                                                          "written image has no zeros"
+
```

Comment:
Wouldn't it be useful to have an assertion which checks that `test_img` and `read_data` hold the same data?

Reply:
Just added it :D